app/models/course/experience_points_record.rb

```diff
@@ -54,4 +55,8 @@ def level_after_update
     current_exp = course_user.experience_points
     course_user.course.level_for(current_exp + exp_changed)
   end
+
+  def set_awarded_at
+    self.awarded_at ||= Time.zone.now
```

Comment:
Ok modified to `before_create` to call only for new records 👍  not sure why I didn't see this earlier >_<